lib/mastodon/emoji_cli.rb

```diff
@@ -96,5 +145,22 @@ def color(green, _yellow, red)
         :red
       end
     end
+
+    def export_file_name(emoji)
+      if options[:shortcode]
+        emoji.shortcode + '.png'
+      else
+        emoji.image_file_name
+      end
+    end
+
+    def add_emoji_to_file(emoji, exported, tar)
+      emoji_file_name = export_file_name(emoji)
+      say("Adding '#{emoji.shortcode}' as '#{emoji_file_name}'...")
+      tar.add_file_simple(emoji_file_name, 0o644, emoji.image_file_size) do |io|
+        io.write Paperclip.io_adapters.for(emoji.image).read
+        exported += 1
```

Comment:
I don't... think this will actually update the `exported` value where you want to use it, since it's just a primitive value you're not updating an object in place, you have a copy of the integer within this method only.

Reply:
You're right. m/ 
After your suggestion with the scope there also is no need to extract that method.